mongodbatlas/data_source_mongodbatlas_online_archive.go

```diff
@@ -0,0 +1,212 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+func dataSourceMongoDBAtlasOnlineArchives() *schema.Resource {
+	singleElement := schemaOnlineArchive()
+	// overwritten to make them read only
+	singleElement["project_id"] = &schema.Schema{
+		Type:     schema.TypeString,
+		Computed: true,
+	}
+
+	singleElement["cluster_name"] = &schema.Schema{
+		Type:     schema.TypeString,
+		Computed: true,
+	}
+
+	singleElement["atlas_id"] = &schema.Schema{
+		Type:     schema.TypeString,
+		Computed: true,
+	}
+
+	return &schema.Resource{
+		Read: dataSourceMongoDBAtlasOnlineArchivesRead,
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"cluster_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"results": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: singleElement,
+				},
+			},
+			"total_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceMongoDBAtlasOnlineArchive() *schema.Resource {
+	return &schema.Resource{
+		Read:   dataSourceMongoDBAtlasOnlineArchiveRead,
+		Schema: schemaOnlineArchive(),
+	}
+}
+
+func schemaOnlineArchive() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		// argument values
+		"project_id": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"cluster_name": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"atlas_id": {
```

Comment:
Kind of a nit but I think atlas_id is confusing.  Since this is the returned id of the online archive perhaps archive_id might make more sense for readability?